src/tsconfig/rulesets/incompatible-options.ts

```diff
@@ -6,5 +6,6 @@ const incompatibleOptions = new RuleSet();
 incompatibleOptions.shouldFail('noEmit', Match.TRUE);
 incompatibleOptions.shouldFail('noLib', Match.TRUE);
 incompatibleOptions.shouldFail('declaration', Match.FALSE);
+incompatibleOptions.shouldFail('emitDeclarationOnly', Match.TRUE);
```

Comment:
the PR description says `emitDeclarationOnly: false` is the incompatible setting, and this reads like `emitDeclarationOnly: true` is incompatible? @mrgrain can you double check that this is what you want before i approve

also til emit and omit are different words